modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredCodec.scala

```diff
@@ -17,47 +17,97 @@
 package io.circe.derivation
 
 import scala.deriving.Mirror
-import scala.compiletime.constValue
+import scala.compiletime.{ constValue, summonInline }
+import scala.quoted.*
 import io.circe.{ Codec, Decoder, Encoder, HCursor, JsonObject }
 
 trait ConfiguredCodec[A] extends Codec.AsObject[A], ConfiguredDecoder[A], ConfiguredEncoder[A]
 object ConfiguredCodec:
-  private def of[A](nme: String, decoders: => List[Decoder[?]], encoders: => List[Encoder[?]], labels: List[String])(
```

Comment:
I would love to be shown wrong on this, but removing this method concerns me.

I know that when you use inline on a private method, (the original `derived` below calls this of), that there is a public accessor created, in this case it should be `inner$of`.

If we have some library, lets call if `A` that used theses macros, and some end user app called `B`.  
What happens if circe is updated to `ofProduct`, built a built with `of`... and `B` updates the version of circe to no longer include `inner#of`?  I feel like we will get a NoSuchMethodError in application `B`?

I'm worried we'll have a similar problem to https://github.com/circe/circe/issues/2089

Reply:
Not sure why mima doesn't catch this situation....

Reply:
Wow I didn't know about that behavior but you're definitely right, I confirmed in a fresh project. I'll put the `of` methods back

Reply:
Oh wait actually, just putting `of` back doesn't fix the issue. The `inline$of` method only exists in the bytecode if `of` is called from an `inline` method, so we could also need to add a dummy `inline` method that still calls it. Given that, what do you think the best course of action is @zarthross?

Reply:
Yeah, this is certainly an awkward part of scala3 inlines... we could try adding a private method with the name `inline$of`?  I'm not sure if scala 3 will allow that, or we could try a dummy inline method as you suggested... we'll just have to try a few things and check the generated bytecode....

Reply:
I ended up adding `inline$of` methods. They can't be `private` otherwise they cause a `java.lang.IllegalAccessError` instead of a binary incompatibility, but I made them `private[derivation]` and marked them as `deprecated`